internal/storage/s3/s3.go

```diff
@@ -0,0 +1,139 @@
+package s3
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	strg "github.com/offen/docker-volume-backup/internal/storage"
+	t "github.com/offen/docker-volume-backup/internal/types"
+	u "github.com/offen/docker-volume-backup/internal/utilities"
+	"github.com/sirupsen/logrus"
+)
+
+type S3Storage struct {
+	*strg.StorageBackend
+	client *minio.Client
+}
+
+// Specific init procedure for the S3/Minio storage provider.
+func InitS3(c *t.Config, l *logrus.Logger, s *t.Stats) (*strg.StorageBackend, error) {
```

Comment:
I think it would be better if such a constructor function would return the interface type and not the struct type. Like this, implementation of the interface is totally up to the package.